.github/workflows/ci.yml

```diff
@@ -1,48 +1,118 @@
 name: CI
-
-on:
+'on':
   push:
     branches:
       - main
   pull_request:
     branches:
       - main
-
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  group: '${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}'
   cancel-in-progress: true
-
 jobs:
   check:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-
-      - name: Install deps
-        run: sudo apt-get install libfmt-dev libgflags-dev
-
       - name: Format
         run: cargo fmt --all -- --check
-
+      
       - name: Clippy
         working-directory: cxx-async
         run: cargo clippy -- -D warnings
-
+  
   build:
-    runs-on: ${{ matrix.os }}
+    runs-on: '${{ matrix.os }}'
     strategy:
       matrix:
         os:
           - ubuntu-latest
-    #          - macos-latest
-    #          - windows-latest
+    
     steps:
       - uses: actions/checkout@v4
-
-      - name: Install dependencies on linux
-        if: matrix.os == 'ubuntu-latest'
-        run: sudo apt-get install libfmt-dev libgflags-dev
-
+      - name: Install dependencies
```

Comment:
I'm considering adding a new job for testing `folly` so that we can test them in parallel.